tensorflow/python/ops/math_ops.py

```diff
@@ -2493,6 +2493,7 @@ def reduce_variance(input_tensor, axis=None, keepdims=False, name=None):
   @end_compatibility
   """
   name = name if name else "reduce_variance"
+  input_tensor = ops.convert_to_tensor(input_tensor)
```

Comment:
It's probably better to put this line inside the scope of the `with ops.name_scope(name)`.  (I.e., move it down one line.)  That way, if you're inspecting the graph produced by this op, it will be clear that the conversion was done as part of the reduce_variance op.

Reply:
@edloper, let me do one thing. Let me add test case as well for this. I haven't tested it thoroughly. Let me do that first. re-requesting you again so that ready-to-pull is removed.

Reply:
+1 please add a test for this so that we can ensure it fixes the problem!